cmd/cstor-pool-mgmt/pool/v1alpha2/status.go

```diff
@@ -0,0 +1,105 @@
+package v1alpha2
+
+import (
+	"strings"
+
+	api "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha2"
+	zfs "github.com/openebs/maya/pkg/zfs/cmd/v1alpha1"
+)
+
+const (
+	// StatusNoPoolsAvailable .. not pools available
+	StatusNoPoolsAvailable = "no pools available"
+	// PoolStatusDegraded .. pool is degraded
+	PoolStatusDegraded = "DEGRADED"
+	// PoolStatusFaulted .. pool is in faulty state
+	PoolStatusFaulted = "FAULTED"
+	// PoolStatusOffline .. pool is in offline state
+	PoolStatusOffline = "OFFLINE"
+	// PoolStatusOnline .. pool is online
+	PoolStatusOnline = "ONLINE"
+	// PoolStatusRemoved .. pool is in removed state
+	PoolStatusRemoved = "REMOVED"
+	// PoolStatusUnavail .. pool is unavailable
+	PoolStatusUnavail = "UNAVAIL"
+)
+
+// GetStatus return status of the pool
+func GetStatus(csp *api.CStorNPool) (api.CStorPoolStatus, error) {
+	var status api.CStorPoolStatus
+
+	ret, err := zfs.NewPoolStatus().
+		WithPool(PoolName(csp)).
+		Execute()
+	if err != nil {
+		return status, err
+	}
+
+	status.Phase = parsePoolStatus(string(ret))
+
+	freeSize, er := getPropertyValue(csp, "free")
+	if er != nil {
+		err = ErrorWrapf(err, "Failed to fetch free size")
+	} else {
+		status.Capacity.Free = freeSize
+	}
+
+	usedSize, er := getPropertyValue(csp, "allocated")
+	if er != nil {
+		err = ErrorWrapf(err, "Failed to fetch used size")
+	} else {
+		status.Capacity.Used = usedSize
+	}
+
+	totalSize, er := getPropertyValue(csp, "size")
+	if er != nil {
+		err = ErrorWrapf(err, "Failed to fetch total size")
+	} else {
+		status.Capacity.Total = totalSize
+	}
+
+	return status, err
+}
+
+// parsePoolStatus parse output of `zpool status` command to extract the status of the pool.
+// ToDo: Need to find some better way e.g contract for zpool command outputs.
+func parsePoolStatus(output string) api.CStorPoolPhase {
```

Comment:
does `zpool dump` command have this pool status value in it?

Reply:
yes, but it might be costly since we are dumping everything related to pool in `zpool dump`.
Instead of `zpool dump`, can we use `zpool get health`
```
root@cstor-pool-7fbn-b589dbfc8-qw6ml:/# zpool get health  cstor-66503a8c-a2d8-11e9-9c62-42010a8001df
NAME                                        PROPERTY  VALUE   SOURCE
cstor-66503a8c-a2d8-11e9-9c62-42010a8001df  health    ONLINE  -
```

Reply:
ok